pkgs/development/python-modules/msprime/default.nix

```
@@ -0,0 +1,60 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, setuptools-scm
+, pythonOlder
+, cython
+, gsl
+, numpy
+, newick
+, tskit
+, demes
+, pytestCheckHook
+, pytest-xdist
+}:
+
+buildPythonPackage rec {
+  pname = "msprime";
+  version = "1.2.0";
+  format = "pyproject";
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-YAJa2f0w2CenKubnYLbP8HodDhabLB2hAkyw/CPkp6o=";
+  };
+
+  nativeBuildInputs = [
+    setuptools-scm
+    cython
+  ];
+
+  propagatedBuildInputs = [
+    gsl
+    numpy
+    newick
+    tskit
+    demes
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
```

```suggestion
    pytest
```
Exceptionally, we need to call `pytest` directly here, not `python -m pytest`. 
[Implementation](https://github.com/NixOS/nixpkgs/blob/37e60e78c8dcbc9d047dd7bed04b45b70aa20387/pkgs/development/interpreters/python/hooks/pytest-check-hook.sh#L53)